src/robot/client.ts

```diff
@@ -589,15 +591,29 @@ export class RobotClient implements Robot {
     return response.getStatusList();
   }
 
-  async streamStatus(resourceNames: ResourceName[], duration: Duration) {
+  streamStatus(
+    resourceNames: ResourceName[],
+    duration: Duration
+  ): ResponseStream<proto.Status[]> {
     const { robotService } = this;
     const request = new proto.StreamStatusRequest();
     request.setResourceNamesList(resourceNames);
     request.setEvery(duration);
-    const response = await promisify<
-      proto.StreamStatusRequest,
-      proto.StreamStatusResponse
-    >(robotService.streamStatus.bind(robotService), request);
-    return response.getStatusList();
+
+    const statusStream = robotService.streamStatus(request);
+    if (statusStream === null) {
```

Comment:
```suggestion
    if (!statusStream) {
```

i'm assuming we want this to behave similar to the logic we have in `promisify` right?

Reply:
ya